scipy/_lib/tests/test__util.py

```diff
@@ -331,6 +331,33 @@ def test_contains_nan_2d(self):
         data4 = np.array([["1", 2], [3, np.nan]], dtype='object')
         assert _contains_nan(data4)[0]
 
+    @array_api_compatible
+    @pytest.mark.parametrize("nan_policy", ['propagate', 'omit', 'raise'])
+    def test_array_api(self, xp, nan_policy):
+        rng = np.random.default_rng(932347235892482)
+        x0 = rng.random(size=(2, 3, 4))
+        x = xp.asarray(x0)
+        x_nan = xp.asarray(x0.copy())
```

Comment:
Oops, I meant `copy(x, xp=xp)`. 1 less `np`->`xp` means 1 less potential device transfer

Reply:
Did you mean save a copy? I don't think we can save a device transfer here, since `x0` is guaranteed to be generated by NumPy on the CPU, but we want it on whatever device `xp` is using. 

It looks like this could save a `copy`, though: if the array already needs to be copied from the CPU to another device, we don't need to copy the NumPy array. In that case, this `copy` function is what I was looking for, because what I really wanted was `xp.asarray(x0, copy=True)` to ensure that exactly one copy is performed.

Reply:
I see. You wrote `copy(x0, xp=xp)` above.